pkgs/by-name/re/readest/package.nix

```
@@ -0,0 +1,100 @@
+{
+  rustPlatform,
+  pnpm_9,
+  cargo-tauri,
+  nodejs,
+  pkg-config,
+  webkitgtk_4_1,
+  wrapGAppsHook3,
+  fetchFromGitHub,
+  gtk3,
+  librsvg,
+  libappindicator,
+  openssl,
+  autoPatchelfHook,
+  makeWrapper,
+  lib,
+  desktop-file-utils,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "readest";
+  version = "0.9.5";
+
+  src = fetchFromGitHub {
+    owner = "readest";
+    repo = "readest";
+    tag = "v${version}";
+    hash = "sha256-JtxW3ngP+oHCE6E55+wUL3ptcJ86UEqj2LFXhDQW6/k=";
+    fetchSubmodules = true;
+  };
+
+  sourceRoot = "${src.name}/apps/readest-app";
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-YmuqMqO9AisSHv7iJK/ElQig/fF/+ngrqSzlvtnX2xc=";
+  };
+
+  pnpmRoot = "..";
+
+  cargoHash = "sha256-2Fx31kVJfQrOeGVtkL1ARCo+REbRKO1JBIZEZO93Lis=";
+
+  cargoRoot = "src-tauri";
+
+  buildAndTestSubdir = cargoRoot;
+
+  postPatch = ''
+    substituteInPlace $cargoRoot/Cargo.toml \
+      --replace-fail '"devtools"' '"devtools", "rustls-tls"'
+    substituteInPlace $cargoRoot/tauri.conf.json \
+      --replace-fail '"createUpdaterArtifacts": true' '"createUpdaterArtifacts": false'
+  '';
+
+  preConfigure = ''
+    chmod -R +w ../..
+  '';
+
+  nativeBuildInputs = [
+    cargo-tauri.hook
+    nodejs
+    pnpm_9.configHook
+    pkg-config
+    wrapGAppsHook3
+    autoPatchelfHook
+    makeWrapper
+  ];
+
+  buildInputs = [
+    webkitgtk_4_1
+    gtk3
+    librsvg
+    libappindicator
+    openssl
+  ];
+
+  preBuild = ''
+    pnpm setup-pdfjs
+  '';
+
+  dontWrapGApps = true;
+
+  postFixup = ''
+    wrapProgram $out/bin/readest \
+      ''${gappsWrapperArgs[@]} \
+      --set-default WEBKIT_DISABLE_DMABUF_RENDERER 1 \
+      --prefix PATH : ${lib.makeBinPath [ desktop-file-utils ]}
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Modern, feature-rich ebook reader";
+    homepage = "https://github.com/readest/readest";
+    mainProgram = "readest";
+    license = with lib.licenses; [ agpl3Plus ];
```

Nit: it doesn't need to be an array
```suggestion
    license = lib.licenses.agpl3Plus;
```